src/pynwb/core.py

```diff
@@ -47,6 +47,19 @@ def _error_on_new_warn_on_construct(self, error_msg: str):
             raise ValueError(error_msg)
         warn(error_msg)
 
+    def _error_on_new_pass_on_construct(self, error_msg: str):
+        """
+        Raise an error when a check is violated on instance creation.
+        When reading from a file, do nothing, ensuring that files with
+        invalid data or deprecated neurodata types can be read.
+        If error_msg is set to None the function will simply return
+        without further action.
+        """
+        if error_msg is None:
+            return
```

Comment:
This behavior is odd to me. When would we use this? 
```suggestion
        """
```

Reply:
I think we can simplify `_error_on_new_warn_on_construct` similarly.

Reply:
Oh, I see. It is used in `image.py`. It's a bit unintuitive there:
```python
        self._error_on_new_warn_on_construct(
            error_msg=self._check_external_file_starting_frame_length()
        )
```
I guess I have a slight preference toward:
```python
error_msg = self._check_external_file_starting_frame_length()
if error_msg:
    self._error_on_new_warn_on_construct(error_msg)
```
But I leave it to you to decide.

Reply:
I agree your suggestion is a little more intuitive - I updated both of the `_error_on_new_warn_on_construct` and `_error_on_new_pass_on_construct` functions